src/Kis/SqlBackend.hs

```diff
@@ -23,14 +25,35 @@ import qualified Data.Text as T
 import Kis.Model
 import Kis.Kis
 
-withInMemoryKis :: KisConfig -> KisClient IO a -> IO a
-withInMemoryKis config client =
-    inMemoryBackend >>= \backend -> runClient backend config client
 
-runClient :: (MonadCatch m, MonadBaseControl IO m, MonadIO m)
-          => SqlBackend -> KisConfig -> KisClient m a -> m a
-runClient backend (KisConfig clock) client =
-    runReaderT client (Kis (handleKisRequest backend) clock)
+-- SqlBackend should be changed to (KisBackend e) when
+-- SqlPool branch is merged
+buildKisWithBackend ::
+    (MonadCatch m, MonadBaseControl IO m, MonadIO m)
+    => SqlBackend
+    -> KisConfig
+    -> Kis m
+buildKisWithBackend backend (KisConfig clock) =
+    Kis (handleKisRequest backend) clock
+
+withInMemoryKis ::
+--    (MonadCatch m, MonadBaseControl IO m, MonadIO m)
+    KisConfig
+    -> (Kis IO -> IO a)
+    -> IO a
+withInMemoryKis config f =
+    inMemoryBackend >>= \backend -> f $ buildKisWithBackend backend config
+
+runClient :: Kis m -> KisClient m a -> m a
+runClient kis client = runReaderT client kis
+
+runSingleClient ::
+--    (MonadCatch m, MonadBaseControl IO m, MonadIO m)
+    KisConfig
+    -> KisClient IO a
+    -> IO a
+runSingleClient config client =
+    withInMemoryKis config $ \kis -> runClient kis client
 
```

Comment:
An diesen zwei Stellen habe ich den Typen auf IO spezialisiert, weil ich in den Tests Probleme hatte, dass er m nicht bestimmen konnte. Ich habs auch nicht wirklich geschafft per Typeannotation zu sagen, was ich in den Tests will. Hast du das schonmal gehabt? Eigentlich könnten hier auch die allgemeinen Typen stehen, wobei WithInMemoryBackend davor auch schon auf IO spezialisiert war (Vielleicht aus dem selben Grund?)